lib/logstash/filters/mutate.rb

```diff
@@ -233,6 +246,21 @@ def remove(event)
   end # def remove
 
   private
+  def select(event)
+    replacement = LogStash::Event.new
+    # Keep the meta data
+    replacement[LogStash::Event::TIMESTAMP] = event[LogStash::Event::TIMESTAMP]
+    replacement[LogStash::Event::VERSION] = event[LogStash::Event::VERSION]
+
+    @select.each do |field|
+      field = event.sprintf(field)
+      next unless event.include?(field)
```

Comment:
can we swap this with the line above, so we can short-circuit the `sprintf` computation if not used.